var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -13,41 +13,83 @@ class Elk(MakefilePackage):
     homepage = 'http://elk.sourceforge.net/'
     url      = 'https://sourceforge.net/projects/elk/files/elk-3.3.17.tgz'
 
-    version('3.3.17', sha256='c9b87ae4ef367ed43afc2d43eb961745668e40670995e8e24c13db41b7e85d73')
-
-    # Elk provides these libraries, but allows you to specify your own
-    variant('blas',   default=True,
-            description='Build with custom BLAS library')
-    variant('lapack', default=True,
-            description='Build with custom LAPACK library')
-    variant('fft',    default=True,
-            description='Build with custom FFT library')
-
-    # Elk does not provide these libraries, but allows you to use them
-    variant('mpi',    default=True,
+    version('7.2.42', sha256='73f03776dbf9b2147bfcc5b7c062af5befa0944608f6fc4b6a1e590615400fc6')
+    version('7.1.14', sha256='7c2ff30f4b1d72d5dc116de9d70761f2c206700c69d85dd82a17a5a6374453d2')
+    version('7.0.12', sha256='9995387c681d0e5a9bd52cb274530b23c0370468b6be86f6c90a6ec445cb8a01')
+
+    # what linear algebra packages to use? the choices are
+    # internal - use internal libraries
```

Comment:
We generally try to avoid internal vendored copies of software or external libraries. It can cause multiple versions of the same library to appear in the link line, or make it more difficult to patch software.

Reply:
Is it a requirement? I'd rather keep it as an option. Besides, internal lapack apparently is needed to support linalg=BLIS